src/repo/mod.rs

```diff
@@ -183,6 +184,10 @@ impl<TRepoTypes: RepoTypes> Repo<TRepoTypes> {
         }
     }
 
+    pub async fn list_blocks(&self) -> Result<Vec<Result<Cid, Error>>, Error> {
+        Ok(self.block_store.list().await?)
+    }
```

Comment:
I was thinking that maybe this should be a passthrough method without the `async` but... It's difficult to say what is the lifetime of the returned value without `async fn` so I guess this is ok for now. Did you try this without `async fn`?